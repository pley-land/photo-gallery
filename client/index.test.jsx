import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PhotoGallery from './components/photogallery';


Enzyme.configure({ adapter: new Adapter() });

describe('<PhotoGallery />', () => {
  test('render PhotoGallery', () => {
    const wrapper = shallow(
      <PhotoGallery />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
