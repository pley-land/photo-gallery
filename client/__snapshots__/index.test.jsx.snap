// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoGallery /> render PhotoGallery 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhotoGallery />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Searchbar"
          id="searchbar"
          src={
            Object {
              "process": [Function],
            }
          }
        />,
        <div
          id="top"
        >
          <img
            alt="mapIMG"
            id="mapIMG"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          <div
            id="photo-gallery"
          >
            <div
              id="photo1"
            >
              <img
                alt="Could be portrait or landscape"
                className="default photo"
                id="img1"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src=""
              />
            </div>
            <div
              id="photo2"
            >
              <img
                alt="Could be portrait or landscape"
                className="selected photo"
                id="img2"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src=""
              />
            </div>
            <div
              id="photo3"
            >
              <span
                className="centered shadow"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <img
                  alt="squares"
                  id="squares"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
                <br />
                See All 3
              </span>
              <img
                alt="Could be portrait or landscape"
                className="default photo"
                id="img3"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src=""
              />
            </div>
          </div>
        </div>,
      ],
      "id": "topshelf",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Searchbar",
          "id": "searchbar",
          "src": Object {
            "process": [Function],
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="mapIMG"
              id="mapIMG"
              src={
                Object {
                  "process": [Function],
                }
              }
            />,
            <div
              id="photo-gallery"
            >
              <div
                id="photo1"
              >
                <img
                  alt="Could be portrait or landscape"
                  className="default photo"
                  id="img1"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src=""
                />
              </div>
              <div
                id="photo2"
              >
                <img
                  alt="Could be portrait or landscape"
                  className="selected photo"
                  id="img2"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src=""
                />
              </div>
              <div
                id="photo3"
              >
                <span
                  className="centered shadow"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <img
                    alt="squares"
                    id="squares"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <br />
                  See All 3
                </span>
                <img
                  alt="Could be portrait or landscape"
                  className="default photo"
                  id="img3"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src=""
                />
              </div>
            </div>,
          ],
          "id": "top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "mapIMG",
              "id": "mapIMG",
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  id="photo1"
                >
                  <img
                    alt="Could be portrait or landscape"
                    className="default photo"
                    id="img1"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />
                </div>,
                <div
                  id="photo2"
                >
                  <img
                    alt="Could be portrait or landscape"
                    className="selected photo"
                    id="img2"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />
                </div>,
                <div
                  id="photo3"
                >
                  <span
                    className="centered shadow"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      alt="squares"
                      id="squares"
                      src={
                        Object {
                          "process": [Function],
                        }
                      }
                    />
                    <br />
                    See All 3
                  </span>
                  <img
                    alt="Could be portrait or landscape"
                    className="default photo"
                    id="img3"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />
                </div>,
              ],
              "id": "photo-gallery",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="Could be portrait or landscape"
                      className="default photo"
                      id="img1"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      src=""
                    />,
                    false,
                  ],
                  "id": "photo1",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Could be portrait or landscape",
                      "className": "default photo",
                      "id": "img1",
                      "onClick": [Function],
                      "onMouseEnter": [Function],
                      "onMouseLeave": [Function],
                      "src": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  false,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Could be portrait or landscape"
                    className="selected photo"
                    id="img2"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />,
                  "id": "photo2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Could be portrait or landscape",
                    "className": "selected photo",
                    "id": "img2",
                    "onClick": [Function],
                    "onMouseEnter": [Function],
                    "onMouseLeave": [Function],
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="centered shadow"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <img
                        alt="squares"
                        id="squares"
                        src={
                          Object {
                            "process": [Function],
                          }
                        }
                      />
                      <br />
                      See All 3
                    </span>,
                    <img
                      alt="Could be portrait or landscape"
                      className="default photo"
                      id="img3"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      src=""
                    />,
                  ],
                  "id": "photo3",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <img
                          alt="squares"
                          id="squares"
                          src={
                            Object {
                              "process": [Function],
                            }
                          }
                        />,
                        <br />,
                        "See All 3",
                      ],
                      "className": "centered shadow",
                      "onClick": [Function],
                      "onMouseEnter": [Function],
                      "onMouseLeave": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "alt": "squares",
                          "id": "squares",
                          "src": Object {
                            "process": [Function],
                          },
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                      "See All 3",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Could be portrait or landscape",
                      "className": "default photo",
                      "id": "img3",
                      "onClick": [Function],
                      "onMouseEnter": [Function],
                      "onMouseLeave": [Function],
                      "src": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Searchbar"
            id="searchbar"
            src={
              Object {
                "process": [Function],
              }
            }
          />,
          <div
            id="top"
          >
            <img
              alt="mapIMG"
              id="mapIMG"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            <div
              id="photo-gallery"
            >
              <div
                id="photo1"
              >
                <img
                  alt="Could be portrait or landscape"
                  className="default photo"
                  id="img1"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src=""
                />
              </div>
              <div
                id="photo2"
              >
                <img
                  alt="Could be portrait or landscape"
                  className="selected photo"
                  id="img2"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src=""
                />
              </div>
              <div
                id="photo3"
              >
                <span
                  className="centered shadow"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <img
                    alt="squares"
                    id="squares"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <br />
                  See All 3
                </span>
                <img
                  alt="Could be portrait or landscape"
                  className="default photo"
                  id="img3"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src=""
                />
              </div>
            </div>
          </div>,
        ],
        "id": "topshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Searchbar",
            "id": "searchbar",
            "src": Object {
              "process": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="mapIMG"
                id="mapIMG"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />,
              <div
                id="photo-gallery"
              >
                <div
                  id="photo1"
                >
                  <img
                    alt="Could be portrait or landscape"
                    className="default photo"
                    id="img1"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />
                </div>
                <div
                  id="photo2"
                >
                  <img
                    alt="Could be portrait or landscape"
                    className="selected photo"
                    id="img2"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />
                </div>
                <div
                  id="photo3"
                >
                  <span
                    className="centered shadow"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      alt="squares"
                      id="squares"
                      src={
                        Object {
                          "process": [Function],
                        }
                      }
                    />
                    <br />
                    See All 3
                  </span>
                  <img
                    alt="Could be portrait or landscape"
                    className="default photo"
                    id="img3"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src=""
                  />
                </div>
              </div>,
            ],
            "id": "top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "mapIMG",
                "id": "mapIMG",
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    id="photo1"
                  >
                    <img
                      alt="Could be portrait or landscape"
                      className="default photo"
                      id="img1"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      src=""
                    />
                  </div>,
                  <div
                    id="photo2"
                  >
                    <img
                      alt="Could be portrait or landscape"
                      className="selected photo"
                      id="img2"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      src=""
                    />
                  </div>,
                  <div
                    id="photo3"
                  >
                    <span
                      className="centered shadow"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <img
                        alt="squares"
                        id="squares"
                        src={
                          Object {
                            "process": [Function],
                          }
                        }
                      />
                      <br />
                      See All 3
                    </span>
                    <img
                      alt="Could be portrait or landscape"
                      className="default photo"
                      id="img3"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      src=""
                    />
                  </div>,
                ],
                "id": "photo-gallery",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="Could be portrait or landscape"
                        className="default photo"
                        id="img1"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        src=""
                      />,
                      false,
                    ],
                    "id": "photo1",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "Could be portrait or landscape",
                        "className": "default photo",
                        "id": "img1",
                        "onClick": [Function],
                        "onMouseEnter": [Function],
                        "onMouseLeave": [Function],
                        "src": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    false,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Could be portrait or landscape"
                      className="selected photo"
                      id="img2"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      src=""
                    />,
                    "id": "photo2",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Could be portrait or landscape",
                      "className": "selected photo",
                      "id": "img2",
                      "onClick": [Function],
                      "onMouseEnter": [Function],
                      "onMouseLeave": [Function],
                      "src": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="centered shadow"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <img
                          alt="squares"
                          id="squares"
                          src={
                            Object {
                              "process": [Function],
                            }
                          }
                        />
                        <br />
                        See All 3
                      </span>,
                      <img
                        alt="Could be portrait or landscape"
                        className="default photo"
                        id="img3"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        src=""
                      />,
                    ],
                    "id": "photo3",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <img
                            alt="squares"
                            id="squares"
                            src={
                              Object {
                                "process": [Function],
                              }
                            }
                          />,
                          <br />,
                          "See All 3",
                        ],
                        "className": "centered shadow",
                        "onClick": [Function],
                        "onMouseEnter": [Function],
                        "onMouseLeave": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "alt": "squares",
                            "id": "squares",
                            "src": Object {
                              "process": [Function],
                            },
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "img",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": "br",
                        },
                        "See All 3",
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "Could be portrait or landscape",
                        "className": "default photo",
                        "id": "img3",
                        "onClick": [Function],
                        "onMouseEnter": [Function],
                        "onMouseLeave": [Function],
                        "src": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
